python_modules/libraries/dagster-dg/dagster_dg/docs.py

```diff
@@ -0,0 +1,168 @@
+import tempfile
+import webbrowser
+from dataclasses import dataclass
+from typing import AbstractSet, Any, Mapping, Optional, Sequence, Union
+
+import markdown
+import yaml
+
+from dagster_dg.context import RemoteComponentType
+
+REF_BASE = "#/$defs/"
+COMMENTED_MAPPING_TAG = "!commented_mapping"
+
+
+@dataclass(frozen=True)
+class CommentedObject:
+    key: str
+    value: Union[Sequence["CommentedObject"], Any]
+    comment: Optional[str]
+    top_level: bool = False
+
+
+def _dereference_schema(
+    json_schema: Mapping[str, Any], subschema: Mapping[str, Any]
+) -> Mapping[str, Any]:
+    if "$ref" in subschema:
+        return json_schema["$defs"].get(subschema["$ref"][len(REF_BASE) :])
+    else:
+        return subschema
+
+
+def _get_commented_obj(
```

Comment:
nit: maybe `commented_object_for_subschema`? slightly unclear from the name that this gets a commented object corresponding to the subschema and not e.g. the overall schema